appshell/appshell_extensions_win.cpp

```diff
@@ -403,6 +403,170 @@ void CloseLiveBrowser(CefRefPtr<CefBrowser> browser, CefRefPtr<CefProcessMessage
     }
 }
 
+
+static BOOL ResolveAppPathCommon(const ExtensionString& lpszKey, const ExtensionString& lpszVal, ExtensionString& appPath)
+{
+    HKEY	keyRoot = NULL;
+    BOOL	result = FALSE;
+    ULONG	regKey(REG_SZ);
+
+    if (::RegOpenKeyEx(HKEY_CLASSES_ROOT, lpszKey.c_str(), 0, KEY_QUERY_VALUE, &keyRoot) == ERROR_SUCCESS)
+    {
+        TCHAR editorPath[255];
+        ULONG fTypeSize = 255;
+
+        if (::RegQueryValueEx(keyRoot, lpszVal.c_str(), NULL, &regKey, (LPBYTE)editorPath, &fTypeSize) == ERROR_SUCCESS)
+        {
+            if (editorPath != L"")
+            {
+                appPath = editorPath;
+                //appPath = appPath.substr(appPath.find_last_of(L"/\\") + 1);
+                int iExe = appPath.find(L".exe");
+                if (iExe == -1)
+                    iExe = appPath.find(L".EXE");
+                if (iExe != -1)
+                    appPath = appPath.substr(0, iExe+4);
+
+                if (appPath[0] == '\"')
+                    appPath = appPath.substr(1);
+
+                result = TRUE;
+            }
+
+
+        }
+        ::RegCloseKey(keyRoot);
+    }
+
+    return result;
+
+}
+
+static BOOL ResolveAppPathFromProgID(const ExtensionString& lpszProgID, ExtensionString& appPath)
+{
+
+    ExtensionString key;
+    key = lpszProgID + L"\\shell\\open\\command";
+
+
+    if (!ResolveAppPathCommon(key, L"", appPath))
+    {
+        key = lpszProgID + L"\\Application";
+        if (ResolveAppPathCommon(key, L"AppUserModelID", appPath)) {
+            appPath = appPath.substr(0, appPath.find(L"_"));
+            return true;
+        }
+        return false;
+    }
+    return true;
+
+}
+
+BOOL GetShellDefaultOpenWithProgPath(const ExtensionString& fileExt, ExtensionString &appPath)
+{
+    HKEY	extKey = NULL;
+    ExtensionString key = fileExt;
+    BOOL	result = FALSE;
+    ULONG	regKey(REG_SZ);
+
+    if (fileExt[0] != '.')
```

Comment:
fileExt can be empty, prefer fileExt.front().

Reply:
if "fileext" is empty, it should return false. added code above.